test/test_validation.py

```diff
@@ -35,6 +35,18 @@ def test_length(self):
         report.addWarning(None, None, 'Fake warning')
         self.assertEqual(2, len(report))
 
+    def test_str(self):
+        # str representation should consist of errors and warnings
+        # all errors precede any warning
+        report = sbol3.ValidationReport()
+        self.assertEqual('', report.__str__())
```

Comment:
Please use `str(report)` throughout this test instead of invoking `report.__str__()` directly. I think it is clearer and more pythonic.

```suggestion
        self.assertEqual('', str(report))
```